### PR TITLE
[Event Hubs] Sample Updates

### PR DESCRIPTION
**Pull request by jsquire**
# Summary 

The focus of these changes is to make some minor enhancements to samples, adding a table of contents to the mocking samples.  A new scenario for mocking the processor has also been added.

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Snippets/Sample08_MockingClientTypesLiveTests.cs [hunk 1]
```diff
@@ -90,5 +91,96 @@ Task processErrorHandler(ProcessErrorEventArgs args)
...
+            Task processEventHandler(ProcessEventArgs args)
+            {
+                // Application-defined code here
+
+                return Task.CompletedTask;
+            }
+
+            // This function simulates a random event being emitted for processing.
+
+            Random rng = new();
```
Comment (christothes) [inline sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Snippets/Sample08_MockingClientTypesLiveTests.cs:115]: not sure if this was a typo
```suggestion
            Random rnd = new();
```

**Review by christothes — APPROVED** [review]

File: sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample11_MockingClientTypes.md [hunk 2]
```diff
@@ -306,7 +317,58 @@ await foreach (PartitionEvent receivedEvent in consumer.ReadEventsFromPartitionA
...
+
+using CancellationTokenSource cancellationTokenSource = new();
+
+IEnumerable<EventData> receivedBatch = await receiver.ReceiveBatchAsync(
+    maximumEventCount: 10,
+    TimeSpan.FromSeconds(1),
+    cancellationTokenSource.Token);
+```
+
+## Querying metadata properties using the `EventHubProducerClient`, `EventHubBufferedProducerClient`, or `EventHubConsumerClient`
```
Comment (m-redding) [inline sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample11_MockingClientTypes.md:371]: I like this title more too!

**Review by m-redding — APPROVED** [review]
I think the new sample is really helpful

File: sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Snippets/Sample08_MockingClientTypesLiveTests.cs (hunk 1 above)
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Snippets/Sample08_MockingClientTypesLiveTests.cs:115]: You got something against (r)andom (n)umber (g)enerator?   😄

File: sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample11_MockingClientTypes.md (hunk 2 above)
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample11_MockingClientTypes.md:371]: I was trolling through other samples and realized this is how we seem to refer to it elsewhere.

**Pull request merged**